code/segmentedtickerplant/stplog.q

```diff
@@ -0,0 +1,303 @@
+// Utilites for periodic tp logging in stp process
+
+// Live logs and handles to logs for each table
+currlog:([tbl:`symbol$()]logname:`symbol$();handle:`int$())
+
+// View of log file handles for faster lookups
+loghandles::exec tbl!handle from currlog
+
+\d .stplg
+
+// Name of error log file
+errorlogname:@[value;`.stplg.errorlogname;`segmentederrorlogfile]
+
+// Create stp log directory
+// Log structure `:stplogs/date/tabname_time
+createdld:{[name;date]
+  $[count dir:getenv[`KDBTPLOG];
+    [.os.md dir;.os.md .stplg.dldir:hsym`$raze dir,"/", string name,"_",date];
+    [.lg.e[`stp;"log directory not defined"];exit]
+  ]
+ };
+
+// Functions to generate log names in one of five modes ///////////////////////////////
+
+logname:@[value;`.stplg.logname;enlist[`]!enlist ()];
+
+// Default stp mode is tabperiod
+// TP log rolled periodically (default 1 hr), 1 log per table
+logname[`tabperiod]:{[dir;tab;p]
+  ` sv hsym[dir],`$raze/[string (.proc.procname;"_";tab;"dv"$p)] except ".:"
+ };
+
+// Standard TP mode - write all tables to single log, roll daily
+logname[`singular]:{[dir;tab;p]
+  ` sv(hsym dir;`$string[.proc.procname],"_",raze[string"dv"$p]except".:")
```

Comment:
why is this function doing it in a different way than the other two? e.g. raze/ in the other two
I think you should do it like this: 
`$string[.proc.procname],"_",(raze string "dv"$p)except ".:" 
And I think the last bit should be a function to create the string of the timestamp that exists on the end of all the log files